app/models/user.rb

```diff
@@ -87,4 +88,17 @@ def send_devise_notification(notification, *args)
   def decorate
     UserDecorator.new(self)
   end
+
+  private
+
+  # used by zxcvbn
+  def weak_words
+    [APP_NAME]
+  end
+
+  # method required by zxcvbn
+  def password_required?
+    confirmable = confirmed? || confirmation_token.present?
+    password.present? && confirmable
+  end
```

Comment:
To clarify, I am complaining about Devise gems that dump everything in the model.